app/javascript/courses/review/components/CoursesReview__Editor.res

```diff
@@ -1251,33 +1251,37 @@ let make = (
-                  <button
-                    onClick={_ => send(ChangeReportVisibility)}
-                    className="inline-flex items-center text-primary-500 px-3 py-2 rounded font-semibold hover:text-primary-700 hover:bg-gray-400 focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 transition">
-                    <span className="hidden md:block pr-3">
-                      {str(
-                        state.showReport
-                          ? t("hide_test_report_button")
-                          : t("show_test_report_button"),
-                      )}
-                    </span>
-                    {
-                      let toggleTestReportIcon = state.showReport
-                        ? "i-arrows-collapse-light"
-                        : "i-arrows-expand-light"
-                      <span className="inline-block w-5 h-5">
-                        <Icon className={"if text-xl " ++ toggleTestReportIcon} />
-                      </span>
-                    }
-                  </button> : React.null}
+                  {SubmissionReport.testReport(report)->Belt.Option.isSome
+                    ? <button
+                        onClick={_ => send(ChangeReportVisibility)}
+                        className="inline-flex items-center text-primary-500 px-3 py-2 rounded font-semibold hover:text-primary-700 hover:bg-gray-400 focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 transition">
+                        <span className="hidden md:block pr-3">
+                          {str(
+                            state.showReport
+                              ? t("hide_test_report_button")
+                              : t("show_test_report_button"),
+                          )}
+                        </span>
+                        {
+                          let toggleTestReportIcon = state.showReport
+                            ? "i-arrows-collapse-light"
+                            : "i-arrows-expand-light"
+                          <span className="inline-block w-5 h-5">
+                            <Icon className={"if text-xl " ++ toggleTestReportIcon} />
+                          </span>
+                        }
+                      </button>
+                    : React.null}
                 </div>
                 {state.showReport
                   ? <div>
                       <p className="text-sm font-semibold mt-4"> {str("Test Report")} </p>
                       <div className="bg-white p-3 rounded-md border mt-2">
                         <MarkdownBlock
-                          profile=Markdown.AreaOfText markdown={SubmissionReport.testReport(report) -> Belt.Option.mapWithDefault("", s => s)}
+                          profile=Markdown.Permissive
+                          markdown={SubmissionReport.testReport(
+                            report,
+                          )->Belt.Option.mapWithDefault("", s => s)}
```

Comment:
[Belt.Option.getWithDefault](https://rescript-lang.org/docs/manual/latest/api/belt/option#getwithdefault) would be the correct choice here.

There appears to be a violation of the _make invalid states impossible_ guideline here, though. Why would there be a `state.showReport=true` if `SubmissionReport.testReport(report)` is `None`? The two are incompatible.

A possible solution is that `state.showReport` is replaced with `state.shownReport: option<CoursesReview__Types.SubmissionReport.t>`.

Reply:
@harigopal I think the confusion here is that `showReport` is the toggle that we use to make the detailed test report visible using the _Show Report_ button:

![image](https://user-images.githubusercontent.com/24313597/163844004-f1866479-6c1c-4470-80dc-c51226593c52.png)

Refactored the block a bit: https://github.com/pupilfirst/pupilfirst/commit/dfe5820f2a3ad22b8921d4f5f123332af6685c47

The button itself is visible if `SubmissionReport.testReport(report)` is `Some`.